test/middleware/spec.resolve-path.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const baseController = require('../helpers/controller');
+const resolvePath = require('../../lib/middleware/resolve-path');
+
+describe('middleware/resolve-path', () => {
+    let BaseController, StubController, controller;
+
+    beforeEach(() => {
+        BaseController = baseController();
+        StubController = resolvePath(BaseController);
+        controller = new StubController();
+    });
+
+    describe('resolvePath', () => {
+
+        it('should return the url value if not a string', () => {
+            expect(controller.resolvePath(null, null)).to.be.null;
+        });
+
+        it('should return am http url as-is', () => {
+            controller.resolvePath(null, 'http://blah').should.equal('http://blah');
+        });
+
+        it('should return am https url as-is', () => {
+            controller.resolvePath(null, 'https://blah').should.equal('https://blah');
+        });
```

Comment:
There is one test for http:// urls and one test for https:// urls.
changed the second description to "should return a secure http url as-is" for better clarity